### PR TITLE
added remaining rspec stuff

### PR DESCRIPTION
**Pull request by troyleach**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -22,7 +22,7 @@ gem 'uglifier', '>= 1.3.0'
 gem 'coffee-rails', '~> 4.2'
 # See https://github.com/rails/execjs#readme for more supported runtimes
 # gem 'therubyracer', platforms: :ruby
-gem 'faker'
+#
```
Comment (troyleach) [inline Gemfile]: delete this ```#```

File: spec/controllers/sessions_controller_spec.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 require 'rails_helper'
+include RequestSpecHelper
```
Comment (troyleach) [inline spec/controllers/sessions_controller_spec.rb]: delete ```include RequestSpecHelper```

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -25,10 +26,12 @@
 # Checks for pending migration and applies them before tests are run.
 # If you are not using ActiveRecord, you can remove this line.
 ActiveRecord::Migration.maintain_test_schema!
+Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }
```
Comment (troyleach) [inline spec/rails_helper.rb:29]: do more research on including helper methods before I merge!

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -25,10 +26,12 @@
...
 
 RSpec.configure do |config|
   # Remove this line if you're not using ActiveRecord or ActiveRecord fixtures
   config.fixture_path = "#{::Rails.root}/spec/fixtures"
+  config.include RequestSpecHelper, type: :request
```
Comment (troyleach) [inline spec/rails_helper.rb:34]: see comment above

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -54,4 +57,26 @@
...
+  end
+end
+
+Shoulda::Matchers.configure do |config|
+  config.integrate do |with|
+    # Choose a test framework:
+    with.test_framework :rspec
+    # Or, choose the following (which implies all of the above):
+    with.library :rails
+  end
```
Comment (troyleach) [inline spec/rails_helper.rb:81]: I think I messed up the ```end```s ???

File: spec/controllers/api/v1/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+include RequestSpecHelper
+
+RSpec.describe Api::V1::UsersController, type: :controller do
+end
```
Comment (troyleach) [inline spec/controllers/api/v1/users_controller.rb]: Add in the two ```create(:user)``` for examples for clarity 

```
create(:user)
create(:admin_user)
```

**Review by troyleach — COMMENTED** [review]
need to fix before merging

File: Gemfile (hunk 1 above)
Comment (troyleach) [reply Gemfile]: done

File: spec/controllers/sessions_controller_spec.rb (hunk 2 above)
Comment (troyleach) [reply spec/controllers/sessions_controller_spec.rb]: done

File: spec/rails_helper.rb (hunk 5 above)
Comment (troyleach) [reply spec/rails_helper.rb:81]: done

File: spec/controllers/api/v1/users_controller.rb (hunk 6 above)
Comment (troyleach) [reply spec/controllers/api/v1/users_controller.rb]: done

File: spec/rails_helper.rb (hunk 3 above)
Comment (kyletgundry) [reply spec/rails_helper.rb:29]: We reviewed helpers and how they are useful when including certain methods globally throughout the app like authentication and how they are called through modules and placed usually in the /lib file.  They are different then models as models are restricted based on the class they are in based on my understanding!

**Pull request merged**